pysilcam/tracking/track.py

```diff
@@ -0,0 +1,417 @@
+        f.visit(objs.append)
+        groups = [obj for obj in objs if isinstance(f[obj], h5py.Group)]
+    return groupstr in groups
+
+
+def subsample_files(datapath, approx_files=2000, offset=int(0)):
+    print('Subsampling files....')
+
+    files = [os.path.join(datapath, f) for f in sorted(os.listdir(datapath)) if
+             f.endswith('.bmp') or f.endswith('.silc')
+             or f.endswith('.silc_mono')]
+    files = files[int(offset):]
+
+    try:
+        times = [f.replace(datapath + '/D', '').replace('.bmp', '') for f in files]
+        times = pd.to_datetime(times)
+    except:
+        try:
+            times = [f.replace(datapath + '/D', '').replace('.silc', '') for f in files]
+            times = pd.to_datetime(times)
+        except:
+            try:
+                times = [f.replace(datapath + '/D', '').replace('.silc_mono', '') for f in files]
+                times = pd.to_datetime(times)
+            except:
+                pass
+
+    t1 = times[0]
+
+    dt = np.abs(times - t1)
+
+    secs = (dt.components.hours * 3600) + (dt.components.minutes * 60) + dt.components.seconds + (
+            dt.components.milliseconds / 1000)
+    image_number = np.arange(0, len(secs))
+
+    base = 10
+    start = 0
+    stop = max(secs)
+    sample_secs = np.logspace(start, np.log(stop) / np.log(base), num=approx_files, base=base)
+    sample_secs = sample_secs[sample_secs < max(secs)]
+    f = interpolate.interp1d(secs, image_number, kind='nearest')
+    sample_idx = f(sample_secs)
+    sample_idx_ = []
+    for i in sample_idx:
+        sample_idx_.append(int(i))
+    sample_idx = np.unique(sample_idx_)
+
+    len(sample_idx)
+
+    files = np.array(np.copy(files))
+    resampled_files = sorted(list(files[sample_idx]))
+
+    print('  ', str(len(resampled_files)), 'files from', str(len(files)))
+
+    return resampled_files
+
+
+def plot_single(datapath):
+    outputpath = make_output_path(datapath)
+    csv_file = (outputpath + '.csv')
+    print('CSV file:', csv_file)
+
+    data = pd.read_csv(csv_file)
+    print(data.columns)
+
+    f, a = plt.subplots(2, 2, figsize=(12, 12))
+
+    plt.sca(a[0, 0])
+    plt.plot(pd.to_datetime(data['Time']), data['ECD [mm]'], '.', color='0.8', alpha=0.1)
+    plt.ylabel('ECD [mm]')
+
+    plt.sca(a[1, 0])
+    plt.plot(pd.to_datetime(data['Time']), data['Speed [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.ylabel('Speed [cm/s]')
+
+    plt.sca(a[0, 1])
+    plt.plot(data['ECD [mm]'], data['Ws [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Ws [cm/s]')
+
+    plt.sca(a[1, 1])
+    plt.plot(data['ECD [mm]'], data['Speed [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Speed [cm/s]')
+
+    plt.show()
+
+
+def load_and_process(tracksfile, PIX_SIZE,
+                     minlength=0, maxlength=1e6, track_length_limit=15):
+    data = pd.read_hdf(tracksfile, 'Tracking/data')
+    tracks = post_process(data, PIX_SIZE,
```

Comment:
F405: 'post_process' may be undefined, or defined from star imports: pysilcam.tracking.silcamtracker  <!-- Automatically posted by Lintly -->